### DISCUSSION TITLE
Best Way To Install Specific Version of SDKMAN

### DISCUSSION THREAD

**Discussion by rgajason** [sdkman/sdkman-cli#1170] [Q&A] [answered]
Hello all,

A recent update to SDKMAN (or one of its dependencies) has broken installs on CentOS 7 due to requirements of newer glibc than is available:

```
/opt/sdkman/libexec/version: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /opt/sdkman/libexec/version)
/opt/sdkman/libexec/version: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /opt/sdkman/libexec/version)
/opt/sdkman/libexec/version: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /opt/sdkman/libexec/version)
/opt/sdkman/libexec/version: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /opt/sdkman/libexec/version)
/opt/sdkman/libexec/version: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /opt/sdkman/libexec/version)
```

This is understandable given the age of EL7 and it going out of support next year.

In the interim while my group is moving to a more recent base OS, I'd like to be able to install a specific (read: older) version of SDKMAN.

One idea I had was to patch the installation script and set the version variables specifically:

```
# Global variables
export SDKMAN_SERVICE="https://api.sdkman.io/2"
export SDKMAN_VERSION="5.16.0"
export SDKMAN_NATIVE_VERSION="0.0.22"
```

...but I don't know the relationship between `SDKMAN_VERSION` and `SDKMAN_NATIVE_VERSION`.

Another option would be to just pull the archive directly from the GH releases, but I don't know how much help the installer is providing that I'd need to re-invent.

Any hints or advice on how to move forward?

**Comment by hgeraldino** [comment, marked as answer]
Maybe it makes sense to check for GLIBC versions when installing the native components. I think for now what you can do, is delete the contents of the ~/.sdkman/libexec folder, and each command will fallback to their bash version.

I'd also suggest posting these sort of questions in our Slack channel, as mentioned in our [contributing guide](https://github.com/sdkman/sdkman-cli/blob/master/CONTRIBUTING.md)

**Reply by helpermethod** [reply]
Maybe for the time being we could add a flag for enforcing the shell versions of the subcommands.

**Reply by rgajason** [reply]
Thanks for the reply and information.  Emptying out the files in the `libexec` folder did, indeed, allow the code to continue to work on EL7 (well, at least it let me install and run `sdk version`).
